scripts/zones/Selbina/npcs/Zaldon.lua

```diff
@@ -512,6 +512,11 @@ local function tradeFish(player, fishId)
     local found   = false
     local sum     = 0
 
+    -- NOTE: We confirm the trade now, and not at the end of the cutscene as normal
+    --     : because the cutscene gives away whether or not the trade was successful
+    --     : or not, and it's possible for players to cheese this trade by force-dc-ing.
+    player:confirmTrade()
```

Comment:
I would recommend moving this to line 519/524

This way, we would make sure the trade is actually valid before nuking it, but before the CS starts